test/test_utils/ec2.py

```diff
@@ -904,3 +909,43 @@ def post_process_mxnet_ec2_performance(connection, log_location):
         return {"Throughput": total / n}
     else:
         raise ValueError("total: {}; n: {} -- something went wrong".format(total, n))
+
+
+@retry(reraise=True, wait=wait_fixed(60), stop=stop_after_attempt(5))
+def install_python_in_instance(context, python_version="3.9"):
+    """
+    Install sagemaker local test dependencies
+    :param context: Invoke Context / Fabric Connection object
+    :param python_version: str python version to install, such as 3.8, 3.9, etc.
+    :return: None
+    """
+    context.run("""ls ~/.pyenv || git clone https://github.com/pyenv/pyenv.git ~/.pyenv""")
+    # Need to configure PATH and PYENV_ROOT changes in alternative location because ~/.bashrc and ~/.profile are
```

Comment:
instead of running several fabric commands like this, do you think it makes more sense to execute an installer shell script? This way if the installation process differs between python versions in the future, we can have versioned shell scripts to install the given version of python

Reply:
Multiple cmds is also fine, but I believe in order to get the most of this we need to use "hide=True" because of an issue in pytest/fabric integration that will print these logs elsewhere if the commands fail

Reply:
Could you elaborate on why we should use `hide=True`?

The reason for using pyenv is that it abstracts away any "special steps" to install various python versions. It should be as straightforward as just `pyenv install <python-version>` to install any version.

I can make it an action item for a future PR to use a pre-configured script, but as of now, I don't see the point.

Reply:
Basically if something in this string of run commands causes a test execution to error, in the stack trace it will likely say something like "Stdout/stderr already printed" and you'll need to search through logs to find the error.

When you use hide=True, it prints the error in line with the stack trace

Reply:
Got it. I can set `hide=True` right now. I'd rather do the script thing later on without the urgency.